Source/Csla.Generators/cs/AutoImplementProperties/Csla.Generator.AutoImplementProperties.CSharp/AutoImplement/Discovery/DefinitionExtractionContext.cs

```diff
@@ -59,13 +60,86 @@ public string GetTypeNamespace(TypeSyntax typeSyntax)
       }
       else
       {
-        
         typeSymbol = _semanticModel.GetSymbolInfo(typeSyntax).Symbol as INamedTypeSymbol;
       }
       if (typeSymbol is null || typeSymbol.ContainingNamespace is null) return string.Empty;
       return typeSymbol.ContainingNamespace.ToString();
     }
 
+    public string GetFullyQualifiedType(TypeSyntax typeSyntax)
+    {
+      INamedTypeSymbol typeSymbol;
+      if (typeSyntax is NullableTypeSyntax nullableTypeSyntax)
+      {
+        typeSyntax = nullableTypeSyntax.ElementType;
+      }
+
+      if (typeSyntax is ArrayTypeSyntax arrayTypeSyntax)
+      {
+        typeSymbol = _semanticModel.GetSymbolInfo(arrayTypeSyntax.ElementType).Symbol as INamedTypeSymbol;
+      }
+      else
+      {
+        typeSymbol = _semanticModel.GetSymbolInfo(typeSyntax).Symbol as INamedTypeSymbol;
+      }
+      if (typeSymbol is null || typeSymbol.ContainingNamespace is null)
+        return string.Empty;
+
+      var typeString = typeSymbol.ToString();
+#pragma warning disable IDE0066 // Convert switch statement to expression, This is more readable than the switch-expression
+      switch (typeString)
```

Comment:
Is there no other way to detect a "native" .NET type?

Perhaps, at least, this chunk of code should be a static method like `IsNativeDotNetType()` or something?

Reply:
I have to double check that. I took this snippet from my source generator and I found no other way at that time. 

I'll move that into some kind of method.

Reply:
Good catch. I got rid of it entirely :) Was not necessary. I'll also update my source gen because it's not necessary there, too xD